core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataReader.java

```diff
@@ -149,6 +151,10 @@ public Factory(FileSystemContext context, WorkerNetAddress address, long blockId
         OpenLocalBlockResponse response = mStream.receive(mDataTimeoutMs);
         Preconditions.checkState(response.hasPath());
         mPath = response.getPath();
+        if (Files.notExists(Paths.get(mPath))) {
```

Comment:
Would you like to use `!Files.exists(Paths.get(mPath))`
```suggestion
        if (!Files.exists(Paths.get(mPath))) {
```

As the following method `notExists` return `false` when encountered IOException

```
    public static boolean notExists(Path path, LinkOption... options) {
        try {
            if (followLinks(options)) {
                provider(path).checkAccess(path);
            } else {
                // attempt to read attributes without following links
                readAttributes(path, BasicFileAttributes.class,
                               LinkOption.NOFOLLOW_LINKS);
            }
            // file exists
            return false;
        } catch (NoSuchFileException x) {
            // file confirmed not to exist
            return true;
        } catch (IOException x) {
            return false;
        }
    }
```